### PR TITLE
Airlocks can now be opened only by moving towards them

### PR DESCRIPTION
**Pull request by DeanUA**
Fixes https://github.com/space-wizards/space-station-14/issues/1024

### CODE REVIEW

**Comment by PJB3005** [comment]
This feels like the kind of thing that should be fixed in the physics engine instead so the collision events don't get raised when, well, there's no collision.

File: Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs [hunk 1]
```diff
@@ -96,6 +98,12 @@ void ICollideBehavior.CollideWith(IEntity entity)
             }
             if (entity.HasComponent(typeof(SpeciesComponent)))
             {
+                if (!entity.TryGetComponent<IMoverComponent>(out var mover)) return;
+                if (!entity.TryGetComponent<ITransformComponent>(out var entityTransform)) return;
+                if (!Owner.TryGetComponent<ITransformComponent>(out var ownerTransform)) return;
```
Comment (ComicIronic) [inline Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs:103]: This shouldn't be necessary. As an entity, `Owner` is guaranteed to have a transform component, accessible through `Owner.Transform`. Same for `entity.Transform`.

Comment (DeanUA) [reply Content.Server/GameObjects/Components/Doors/ServerDoorComponent.cs:103]: @ComicIronic thanks a lot, I'll keep that in mind for the future. I think there is no point fixing it here, as PJB says this should be fixed in physics.

**Comment by ComicIronic** [comment]
I don't think this is a physics engine problem, unless you want to really change the way collisions are raised. In my testing, I was only able to get the issue to work by actually colliding with the door.
collision
I think a collision event is being raised because collision is happening. If we want doors to only open for certain directions of collision, this might be the way to go.

**Comment by PJB3005** [comment]
Yeah but the real weird thing is how I could walk into the airlock by just heading straight at it, stop before it triggers a collision, then *walk the other way* and actually trigger a collision. Something's messed up with that either way.

**Comment by Clyybber** [comment]
> Fixes #1023
I think you meant 1024 :)

**Comment by DeanUA** [comment]
>I think you meant #1024 :)

Yeah, sorry for the wrong link, corrected.

**Comment by PJB3005** [comment]
I merged this but added a comment that it *is* merely a workaround.

**Pull request merged**